doc/redis-sentinel-tls.md

```diff
@@ -0,0 +1,688 @@
+        - name: redis-config-volume
+          configMap:
+            name: redis-config-redis
+        - name: redis-tls-volume
+          secret:
+            secretName: redis-tls-secret
+EOF
+```
+
+
+### Install Sentinel with TLS support
+
+#### Sentinel Service
+
+```yaml
+cat << EOF | oc create -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-sentinel
+spec:
+  selector:
+    app: redis-sentinel
+  ports:
+    - name: redis 
+      protocol: TCP
+      port: 26379 
+      targetPort: 26379 
+    - name: redis-tls
+      protocol: TCP
+      port: 26380 
+      targetPort: 26380
+  clusterIP: None  # This makes the service headless
+EOF
+```
+
+#### Sentinel ConfigMap
+  
+```yaml
+cat << EOF | oc create -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-sentinel-config 
+data:
+  sentinel.conf: |
+    port 0
+    tls-port 26380
+    tls-cert-file /etc/redis/certs/redis-sentinel.crt
+    tls-key-file /etc/redis/certs/redis-sentinel.key
+    tls-ca-cert-file /etc/redis/certs/ca.crt
+    tls-auth-clients yes 
+    tls-replication yes
+
+    #sentinel monitor mymaster redis.3scale-test.svc.cluster.local 6380 2
+    sentinel monitor mymaster  172.30.115.228 6380 2
+     
+    sentinel down-after-milliseconds mymaster 5000
+    sentinel failover-timeout mymaster 10000
+    sentinel parallel-syncs mymaster 1
+EOF
+```
+**Notes** 
+- please make sure use correct Redis service IP in option (example): `sentinel monitor mymaster  172.30.115.228 6380 2`. **Better - to use DNS**. We used IP for testing.
+- if `port 0` is set (as in this example) - the default Sentinel non-TLS port `26379`  will be desabled, and Sentinel will use only TLS communication. 
+- If you will define both `port 26379` (default) and `tls-port 26380`   - Sentinel will use both TLS and non TLS communications. We found that in logs it looks like Sentinel has two replicas that connect to Redis, TLS and non-Tls (details are not provided here).
+
+
+#### Sentinel StatefulSet
+
+```yaml
+cat << EOF | oc create -f -
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: redis-sentinel
+spec:
+  serviceName: "redis-sentinel"
+  replicas: 3
+  selector:
+    matchLabels:
+      app: redis-sentinel
+  template:
+    metadata:
+      labels:
+        app: redis-sentinel
+    spec:
+      initContainers:
+        - name: copy-sentinel-config
+          image: quay.io/fedora/redis-6
+          command: ["/bin/sh", "-c", "cp /etc/redis/sentinel.conf /mnt/config/sentinel.conf"]
+          volumeMounts:
+            - name: redis-config-volume
+              mountPath: /etc/redis/sentinel.conf
+              subPath: sentinel.conf  # Read-only mount from ConfigMap
+            - name: sentinel-config-writable
+              mountPath: /mnt/config  # Writable volume for copying the file
+      containers:
+        - name: redis-sentinel
+          #image: quay.io/fedora/redis-6
+          image: redis:7
+          ports:
+            - containerPort: 26379
```

Comment:
In above cm we only have tls-port 26380 - do we need 26379 as well here?

Reply:
non-TLS port can be useful for debugging in Testing environment. 
Thank you